code/_onclick/item_attack.dm

```diff
@@ -29,13 +29,98 @@ avoid code duplication. This includes items that may sometimes act as a standard
 /obj/item/proc/pre_attack(atom/a, mob/user, var/params)
 	return
 
+//Handles double tact weapons, returns TRUE if attack is to be carried out, FALSE otherwise
+/obj/item/proc/double_tact(mob/user)
+	if(doubletact)
+		if(!(ready))
+			user.visible_message(SPAN_DANGER("[user] raises \his [src]"))
+			ready = TRUE
+			var/obj/effect/effect/melee/alerts/A = new()
+			user.vis_contents += A
+			qdel(A)
+			var/endtime = world.time + (10 SECONDS)
+			while(world.time < endtime)
+				sleep(1)
+				if(!(ready))
+					user.vis_contents -= A
+					return FALSE
+				if(!(is_equipped()))
+					ready = FALSE
+					user.vis_contents -= A
+					return FALSE
+			user.visible_message(SPAN_DANGER("[user] lowers \his [src]"))
+			ready = FALSE
+			user.vis_contents -= A
+			return FALSE
+		else
+			ready = FALSE
+			return TRUE
+	else
+		return TRUE
```

Comment:
It says 10 Seconds on the docs. Your suggestion does not operate in the same way as my code, completely discarding the checks during the 10 seconds that the weapon stays ready. Moreover, it is best to delete A straightaway, since it will be added to viscontents already and serves zero purpose.

Reply:
So you want to
1) Create an object
2) Add it to some list
3) Delete that object
4) Wait 10 seconds
5) Remove it from some list
Is that right?

As for 10 seconds moment, that's insanely long and should be couple of seconds maximum, no point in running 9000 checks during that time.

Reply:
That still doesn't check for dropping the item and such. The real time checks are in there for a reason. I have consulted other coders on this implementation, and I do not see a reason to change it.